File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
dx, dy = [1, 0, -1, 0], [0, 1, 0, -1]

def solution(board, h, w):
    answer = 0
    n = len(board)
    for i in range(4):
        nx = h + dx[i]
        ny = w + dy[i]
        if 0 <= nx < n and 0 <= ny < n:
            if board[h][w] == board[nx][ny]:
                answer += 1
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    
    int[] dx = {1, 0, -1, 0}, dy = {0, 1, 0, -1};
    
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        int n = board.length;
        for (int i = 0; i < 4; i++) {
            int nx = h + dx[i];
            int ny = w + dy[i];
            if (nx >= 0 && nx < n && ny >= 0 && ny < n) {
                if (board[h][w].equals(board[nx][ny])) {
                    answer += 1;
                }
            }
        }
        return answer;
    }
}